File: Challenge2/model/reservation.py
from marshmallow import post_load
from Challenge2.model.event import EventActivities, EventSchema



class Reservation(EventActivities):
    def __init__(self, user, event,startTime,endTime):
        super(Reservation, self).__init__(user, event,startTime,endTime)

    def __repr__(self):
        return '<Reservation(name={self.user!r})>'.format(self=self)


class ReservationSchema(EventSchema):
    @post_load
    def make_reservation(self, data, **kwargs):
        return Reservation(**data)

File: Back_End_Challenge/src/main/java/com/example/reservationsapi/model/Reservation.java
package com.example.reservationsapi.model;

import java.time.Instant;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Reservation {
    
    private String user;
    private String event;
    private Instant startTime;
    private Instant endTime;
}
